plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -122,6 +142,10 @@
 
     private final AggregateFunctionRewriter aggregateFunctionRewriter;
 
+    private static final int SQLSERVER_MAX_SUPPORTED_TIMESTAMP_PRECISION = 7;
+
+    private static final TimeZone UTC_TIME_ZONE = TimeZone.getTimeZone(ZoneId.of("UTC"));
```

Comment:
sadly, `TimeZone` is not immutable (too). Technically speaking, you should `.clone()` every time you use it. (i consider this dark ages in Java history...)

Reply:
I replaced the property with calls to `TimeZone.getTimeZone("UTC")`. Using a string argument avoids some processing and it does call `clone()` internally before returning a result.

Yeah I did the research on `java.sql.Timezone` vs the new classes ;-)